tftpsync/susemanager-tftpsync-recv/add.wsgi

```diff
@@ -140,3 +144,10 @@ def application(environ, start_response):
     return [content.encode()]
 
 
+def application(environ, start_response):
+    try:
+        _application(environ, start_response)
+    finally:
+        # Cleanup tmpdir to cleanup temporary files in case server times out
+        tempdir = os.path.join(CFG.TFTPBOOT, "tmp")
+        shutil.rmtree(tempdir)
```

Comment:
Could it happen that this cleanup runs while another execution is being performed? In the case of several files being pushed at the same time?

Reply:
Yes, that indeed is a problem. But we cannot prevent that from happening easily as we require the cleanup due to SUMA server timeouts that might occur. Since this is a temporary directory, this should only require a second `cobbler sync` and all the necessary files should be present.

My assumption is that during the next `cobbler sync` not the exact same file is running into the problem you just described. If that assumption would be wrong, we would run into a loop.

Reply:
@meaksh Would you be okay if we clean up files that are older than 12h? A `cobbler sync` should never take longer than an hour in ideal conditions so we should have plenty of time to finish the sync up with an 11h buffer.

Reply:
IIUC this cleanup would run on every file that is pushed to the server, and not once per each cobbler sync execution. If that is the case, I wouldn't run this cleanup (even if only for 12h files) on every single file pushed.

What if we create another wsgi script, that is triggered at the end of `cobbler sync` execution, just in order to perform this cleanup of older files after `cobbler sync` execution finished?

Reply:
That sounds very reasonable imho as it is backwards compatible and would solve the problem from the date the submission is integrated onwards. However, due to the containerization efforts, I would say that a complete refactor of all endpoints is then a more reasonable thing to do as `wsgi` scripts in containers are not so "nice".

Reply:
I have found a better solution for the future:

We can introduce signal the sync with a UUID that we send over the wire. That would identify clearly the `cobbler sync` that is currently writing to disk. That also gives us the advantage that we can log those uuids on the proxy side to gain an increased insight in to the data flow.